evm-contracts/src/v0.6/dev/Flags.sol

```diff
@@ -0,0 +1,108 @@
+pragma solidity 0.6.6;
+
+
+import "./SimpleReadAccessController.sol";
+import "./AccessControllerInterface.sol";
+
+
+/**
+ * @title The Flags contract
+ * @notice Allows flags to signal to any reader on the access control list.
+ * The owner can set flags, or designate other addresses to set flags. The
+ * owner must turn the flags off, other setters cannot.
+ */
+contract Flags is SimpleReadAccessController {
+
+  AccessControllerInterface public flaggingAccessController;
+
+  mapping(address => bool) private flags;
+
+  event FlagOn(
+    address indexed subject
+  );
+  event FlagOff(
+    address indexed subject
+  );
+  event SetterEnabled(
+    address indexed setter
+  );
+  event SetterDisabled(
+    address indexed setter
+  );
+
+  constructor(
+    address acAddress
+  )
+    public
+  {
+    setFlaggingAccessController(acAddress);
+  }
+
+  /**
+   * @notice read the warning flag status of a contract address.
+   * @param subject The contract address being checked for a warning.
+   */
+  function getFlag(address subject)
+    public
+    view
+    checkAccess()
+    returns (bool)
+  {
+    return flags[subject];
+  }
+
+  /**
+   * @notice allows owner to enable the warning flags for mulitple addresses.
+   * @param subjects List of the contract addresses whose flag is being raised
+   */
+  function setFlagsOn(address[] calldata subjects)
+    external
+    returns (bool)
+  {
+    require(msg.sender == owner ||
```

Comment:
Here I suggested using a modifier because of the or condition in the `require` statement. The or condition makes it not so simple anymore.

`_onlyWithWriteAccess` reads a lot better to me than:

```
require(msg.sender == owner || setters[msg.sender], "Only callable by enabled setters");
```

And it's even more complicated in the new version:

```
require(msg.sender == owner ||
  flaggingAccessController.hasAccess(msg.sender, msg.data),
  "No access");
```

If I need to know the details, I could inspect the `_onlyWithWriteAccess` modifier, but here I am bombarded with the details every time I read the `setFlagsOn`. I would argue that those details are not so important when reading the function, and it's enough for me to just know there is some write access modifier to check if interested.

These are just my 2 cents. If our general convention is to avoid modifiers in most cases, then I would not like to introduce one just for this function. So please decide on what you think is best 👍

Reply:
We tend to avoid modifiers (unless the code is used in lots of places) to make it easier to follow the precise logic.
I personally would vote for keeping it as is.

Reply:
I've broken up the require statement to at least make it more readable for now.